### PR TITLE
feat: add medusa-react

### PR DESCRIPTION
**Pull request by zakariaelas**
### What

- adds `medusa-react`
- refactors return reasons settings to use `medusa-react` admin hooks
- refactors nested return reasons to showcase usage with query keys and query invalidation
- adds a `useToaster` hook to move away from the `useMedusa` hook

### How

- installs the canary version of `medusa-react`
- uses the `useAdminCreateReturnReason`, `useAdminUpdateReturnReason`, `useAdminDeleteReturnReason`, `useAdminReturnReason`, and `useAdminReturnReasons` hooks to create, update, delete, list, and fetch return reasons
- extracts toaster capability we have in `useMedusa` to another independent `useToaster` hook

### Why

- Sets the required configuration to start consuming `medusa-react` hooks in various admin domains
- Adds a concrete example to how the hooks can be used in our codebase

### CODE REVIEW

File: gatsby-browser.js [hunk 1]
```diff
@@ -5,22 +5,43 @@ import { AccountProvider } from "./src/context/account"
 import { InterfaceProvider } from "./src/context/interface"
 import { ThemeProvider as Provider } from "./src/theme"
 import { ToastProvider } from "react-toast-notifications"
+import { MedusaProvider } from "medusa-react"
+import { QueryClient } from "react-query"
 
 //export const shouldUpdateScroll = ({ routerProps: { location } }) => {
 //  window.scrollTo(0, 0)
 //  return false
 //}
```
Comment (olivermrbl) [inline gatsby-browser.js:14]: nit: Might as well clean this up 😄

File: src/domain/settings/return-reasons/create-edit-return-reason/index.js [hunk 2]
```diff
@@ -193,4 +200,34 @@ const ReturnReasonModal = ({
...
+
+const useDeleteNestedReturnReason = (id, parentId) => {
+  const queryClient = useQueryClient()
+
+  return useAdminDeleteReturnReason(id, {
+    onSuccess: () => {
+      queryClient.invalidateQueries(adminReturnReasonKeys.detail(parentId))
+    },
+  })
+}
```
Comment (olivermrbl) [inline src/domain/settings/return-reasons/create-edit-return-reason/index.js:231]: Very nice! 😍

**Review by olivermrbl — APPROVED** [review]
LGTM. Great work Zak, this is gonna be so nice to work with! 👍 

Couple of comments.

File: gatsby-browser.js [hunk 3]
```diff
@@ -5,22 +5,43 @@ import { AccountProvider } from "./src/context/account"
...
 
+const STORE_URL = process.env.GATSBY_STORE_URL || "http://localhost:9000"
```
[the remark this answers is not in the document]
Comment (zakariaelas) [reply gatsby-browser.js:16]: @olivermrbl if I understood correctly, the only suggested change is to rename the env variable right? If yes, then I agree, I think `MEDUSA_BACKEND_URL` is more accurate than `STORE_URL` :smile:

[the remark this answers is not in the document]
Comment (olivermrbl) [reply gatsby-browser.js:16]: Yes, and backward compatibility 👍  @zakariaelas

File: gatsby-browser.js [hunk 4]
```diff
@@ -1,26 +1,29 @@
 import React from "react"
-import { ToastProvider } from "react-toast-notifications"
-import "./src/assets/styles/global.css"
```
[the remark this answers is not in the document]
Comment (zakariaelas) [reply gatsby-browser.js:3]: crap :fearful: .. well spotted! thanks Kasper!

**Pull request merged**